### PR TITLE
cpio on plan9: convert symlinks to binds

### PR DESCRIPTION
**Pull request by rminnich**
Plan 9 does not have symlinks.

"Symlinks are a botch" -- David Korn

"But they're so damn useful" -- everyone else

Anyway, for now, if cpio sees a symlink, it will now do a
bind by creating an empty file and binding it to the symlink
target (a.k.a. "contents of the cpio record")..

It's the best that can be done at present. It is only
useful when unpacking a cpio into a ramfs. But Harvey
needs that.

Signed-off-by: Ronald G Minnich <rminnich@gmail.com>

### CODE REVIEW

File: pkg/cpio/fs_plan9.go [hunk 1]
```diff
@@ -100,6 +104,22 @@ func CreateFileInRoot(f Record, rootDir string, forcePriv bool) error {
...
+			return err
+		}
+		if err := ioutil.WriteFile(f.Name, []byte{}, 0444); err != nil {
+			return err
+		}
+		content := string(b)
+		if !filepath.IsAbs(content) {
+			content = filepath.Join(dir, content)
+		}
+		if err := syscall.Bind(f.Name, content, syscall.MREPL); err != nil {
```
Comment (gmacd) [inline pkg/cpio/fs_plan9.go:119]: Would be worth adding a plan9-specific flag at some point to decide whether to bind or duplicate the file.  This works well for now though.

**Review by gmacd — APPROVED** [review]

File: pkg/cpio/fs_plan9.go (hunk 1 above)
Comment (rminnich) [reply pkg/cpio/fs_plan9.go:119]: I'ld like to wait until we need it.

**Comment by rminnich** [comment]
btw the reason to not deref is that if you deref, you'll end up with 50 to 100 6M files ... I realize dram is cheap, but there's no reason to be dumb :)

**Review by fhs — APPROVED** [review]

**Comment by rminnich** [comment]
This is too fragile. If the symlinks appear in the archive before the symlink target, all hell breaks loose. Everything we might do to mitigate this just gets kludgier. Time to do something better.

**Pull request closed without merging**